### ISSUE TITLE
False positive for `Layout/FirstArgumentIndentation` and `Layout/ClosingParenthesisIndentation` cops

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#13790]
## Actual behavior

I have:

```ruby
RSpec.describe SecurityMailer do
  subject :result do
    mailer.with(
            recipient: email,
          )
          .change_email_request
  end

  pending 'add something'
end
```

I run rubocop with -A option and I got:
```ruby
RSpec.describe SecurityMailer do
  subject :result do
    mailer.with(
      recipient: email,
    )
          .change_email_request
  end

  pending 'add something'
end
```

## Rubocop

```console
$ rubocop -V
1.71.0 (using Parser 3.3.6.0, rubocop-ast 1.37.0, analyzing as Ruby 2.7, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.21.0
  - rubocop-factory_bot 2.26.1
  - rubocop-performance 1.23.1
  - rubocop-rails 2.29.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 3.4.0
  - rubocop-rspec_rails 2.30.0
  - rubocop-thread_safety 0.6.0
```

**Comment by ydakuka** [comment]
Default config.

I've written tests in `rubocop/spec/` to reproduce the false positives.

1.
```ruby
RSpec.describe RuboCop::Cop::Layout::FirstArgumentIndentation, :config do
  let(:cop_config) { { 'EnforcedStyle' => style } }

  let(:other_cops) { { 'Layout/IndentationWidth' => { 'Width' => indentation_width } } }

  context 'when EnforcedStyle is special_for_inner_method_call_in_parentheses' do
    let(:style) { 'special_for_inner_method_call_in_parentheses' }
    let(:indentation_width) { 2 }

    it 'registers an offense' do
      expect_offense(<<~RUBY)
        mailer.with(
                recipient: email,
                ^^^^^^^^^^^^^^^^ Indent the first argument one step more than the start of the previous line.
              )
              .change_email_request
      RUBY

      expect_correction(<<~RUBY)
        mailer.with(
          recipient: email,
              )
              .change_email_request
      RUBY
    end
  end
end
```

2.
```ruby
RSpec.describe RuboCop::Cop::Layout::ClosingParenthesisIndentation, :config do
  it 'registers an offense' do
    expect_offense(<<~RUBY)
      mailer.with(
        recipient: email,
            )
            ^ Indent `)` to column 0 (not 6)
            .change_email_request
    RUBY

    expect_correction(<<~RUBY)
      mailer.with(
        recipient: email,
      )
            .change_email_request
    RUBY
  end
end
```

**Comment by dvandersluis** [comment]
This appears to be working as expected.

Based on the description in `Layout/FirstArgumentIndentation` of `EnforcedStyle: special_for_inner_method_call` (which `EnforcedStyle: special_for_inner_method_call_in_parentheses` is the same as when there isn't an inner call):

> The first argument should normally be indented one step more than the preceding line, but if it's an argument for a method call that is itself an argument in a method call, then the inner argument should be indented relative to the inner method.

In your example, the first argument to `with` is not a nested method call, therefore `recipient: email` should be indented one step further than the `mailer.with` line. 

Then, looking at `Layout/ClosingParenthesisIndentation`'s documentation:

> Scenario 1: When First Parameter Is On Its Own Line
> when first param is on a new line, right paren is *always*  outdented by IndentationWidth

In this case, the first parameter is on its own line, so the closing parentheses should be one indentation step less than the first parameter, and your code is correctly autocorrected to:

```ruby
RSpec.describe SecurityMailer do
  subject :result do
    mailer.with(
      recipient: email
    )
          .change_email_request
  end

  pending 'add something'
end
```

Note that `Layout/RedundantLineBreak` registers an offense here too which would further rewrite the code if enabled.

I think `EnforcedStyle: consistent_relative_to_receiver` is the configuration that you want here but it does not properly consider method chains right now.

**Comment by dvandersluis** [comment]
There is an existing test for this, added in #5699:

https://github.com/rubocop/rubocop/blob/fe802d9a3b4dca31d1952d77205d2f3546882e93/spec/rubocop/cop/layout/first_argument_indentation_spec.rb#L713-L720

I'm not sure what the rationale was (@bbatsov, any ideas?) but I think either this is wrong (although even if it is we probably can't change it because it'd now be expected behaviour for users of this style), or it'd make sense to have another style (although naming that will be tricky).
